src/mlir_build.cpp

```diff
@@ -95,10 +95,7 @@ void MiirGenBin(const std::string& params, std::vector<char>& buffer)
 int MiirGetKernelCount(const std::string& params)
 {
     AutoMiirHandle handle(params);
-    const auto kernel_count = miirGetKernelCount(handle());
-    if(kernel_count < 1)
-        MIOPEN_THROW("miirGetKernelCount invalid count: " + std::to_string(kernel_count));
-    return kernel_count;
+    return miirGetKernelCount(handle());
```

Comment:
```suggestion
    const auto n = miirGetKernelCount(handle());
    return n < 0 ? 0 : n;
```
This is necessary if `miirGetKernelCount` may return negative value (otherwise `IsValid` may misbehave). Or rocMLIR APL has changed in 5.6?

Reply:
Thanks Artem! More reasonable than just returning -1 as is. 

This API behavior didn't change since it come out. It has been always like this since it get exposed. The reason why I have just discovered this is due to MIOpen ctest starts to do tuning for smoke tests, and having throw in the log make the ctest to fail. That's how I realize this issue exists.

Reply:
@jerryyin Well, actually I was wrong when I said that _"...(otherwise `IsValid` might not behave correctly)."_ That would have worked correctly. On the other hand, now the code looks better because the `MiirGetKernelCount` wrapper returns the mathematically correct number of kernels.